notes/git-managing-branches.md

```diff
@@ -0,0 +1,59 @@
+**Managing branches**
+Branches purpose is to isolate your development work from other branches in the repository. For example, you could use a branch to develop a new feature or fix a bug.
+
+You always create a branch from an existing branch. Typically you create a branch from the default branch of your repository. You can then work on this new branch in isolation from changes that other people are making to the repository.
+
+Once you're satisfied with your work, you can create a pull request to merge your changes in the current branch into another branch.
+
+**1. Creating a Branch**
+The first new branch is based on the default branch. In case that other branches exist - you can decide which to choose as base for new branch. To do that be sure to be checked out on a branch you want to use as base and then run `git branch <new branch>`. To checkouut to a new one run `git checkout <new branch>`.
```

Comment:
remember to add all the git bash command in the correct format

Reply:
```bash
git push -u origin <new branch>
``` 
added such format  for all single-line command examples